mycelite/src/replicator.rs

```diff
@@ -46,71 +44,36 @@ impl Replicator {
     }
 
     pub fn spawn(mut self) -> ReplicatorHandle {
-        let (local_loop_tx, mut local_loop_rx) = channel();
-        let (remote_loop_tx, mut remote_loop_rx) = channel();
-        let mut local_loop_clone = local_loop_tx.clone();
-        let read_only = self.read_only;
-        let config = Arc::clone(&self.config);
-        let local_loop_h = Some(std::thread::spawn(move || {
-            self.enter_local_loop(&mut local_loop_rx)
-        }));
-        let remote_loop_h = match !read_only {
-            true => None,
-            false => Some(std::thread::spawn(move || {
-                Self::enter_remote_loop(&mut local_loop_clone, &mut remote_loop_rx, config)
-            })),
-        };
-        ReplicatorHandle::new(local_loop_tx, remote_loop_tx, local_loop_h, remote_loop_h)
+        let (tx, mut rx) = channel();
+        let local_h = Some(std::thread::spawn(move || self.enter_loop(&mut rx)));
+        ReplicatorHandle::new(tx, local_h)
     }
 
     /// local loop
     ///
     /// listens for notifications pulls/pushes snapshots, restores underlying database to latest
     /// snapshot
-    fn enter_local_loop(&mut self, rx: &mut Receiver<Message>) {
+    fn enter_loop(&mut self, rx: &mut Receiver<Message>) {
         loop {
-            if !self.read_only {
-                self.maybe_push_snapshots().ok();
+            match self.read_only {
+                true => {
+                    match self.maybe_pull_snapshots() {
+                        Ok((last, new)) if last < new => {
+                            self.restore_latest_snapshot().ok();
+                        }
+                        Ok(_) => (),
+                        Err(_e) => (),
+                    };
+                }
+                false => {
+                    self.maybe_push_snapshots().ok();
+                }
             }
-            match rx.recv_timeout(std::time::Duration::from_secs(5)) {
+            match rx.recv_timeout(std::time::Duration::from_secs(1)) {
```

Comment:
Do you think a constant would be clearer?

Reply:
nah, better replicator in near future?

Reply:
Agreed 👍